Prevent zero-priced CFD offers

klochowicz opened the pull request:
Simplify the UI code and disable the button for posting a new offer if there is
no price set.

klochowicz committed:
Disable posting new offers in the maker when there is no order price

thomaseizinger commented on frontend/src/MakerApp.tsx:
```diff
@@ -187,18 +185,6 @@ export default function App() {
-    }
-
-    if (!hasEnteredPrice) {
-        return format((priceInfo.ask * SPREAD).toString());
```
If you delete this, we are no longer calculating a default price using the spread.

klochowicz replied:
there was no default price using the spread. It was displayed, but not used anywhere... that was the bug in question.

klochowicz replied:
is there a way of triggering the "refresh" button when the `priceInfo` becomes available? That would solve the actual problem in a much nicer way IMHO.

thomaseizinger commented on frontend/src/MakerApp.tsx:
```diff
@@ -187,18 +185,6 @@ export default function App() {
-        return format((priceInfo.ask * SPREAD).toString());
-    }
-
-    return format(orderPrice);
```
We are no longer calling `format` on the price now, is that intentional?

klochowicz replied:
fixed (as per the fixup commit) - added format when the button refreshes.

thomaseizinger reviewed (changes requested):
Two questions.

klochowicz committed:
Allow auto-refreshing price suggestions in the maker

If the switch is enabled, the maker gets price suggestions based on the current
ask price on Bitmex + 3% spread.

Auto-refresh turns off whenever user modifies the field manually, and can be
re-enabled later if desired.

Fixes #214

thomaseizinger commented on frontend/src/MakerApp.tsx:
```diff
@@ -107,19 +112,18 @@ export default function App() {
                             />
+                            <HStack>
+                                <Switch
+                                    id="auto-refresh"
```
This can go away if you are not using it for attaching a label to the component.

klochowicz replied:
I find it still useful for conveying the purpose of the switch (it's self-explanatory now, without going into the details of the callback)

da-kami reviewed (approved):
noice!
